src/spi.rs

```diff
@@ -1298,9 +1638,124 @@ fn polling_transmit(
         length: (transaction_length * 8) as _,
         rxlength: (rx_length * 8) as _,
         ..Default::default()
-    };
+    }
+}
+
+fn spi_transmit(
+    handle: spi_device_handle_t,
+    transaction: &mut spi_transaction_t,
```

Comment:
`_keep_cs_active` would then need to be the caller's responsibility.

Reply:
Did you mean `spi_transmit_async`? Regardless, I do not understand your point. `_keep_cs_active` **IS** already the caller's responisibility.

Reply:
> Did you mean `spi_transmit_async`?

I actually mean both. They both use interrupts so queueing can be done.

> `_keep_cs_active` IS already the caller's responsibility.

I mean the caller would be doing the `transaction.flags |= KEEP_CS_ACTIVE`, instead of doing it down here.
If it still doesn't make sense, just ignore it. It should make sense once you start passing iterators.

Reply:
Why don't we just leave the transaction queueing for a subsequent PR? And hence the changes you suggest would only be implemented then?

Reply:
I'm not suggesting to implement transaction queueing in this PR.
I'm suggesting that an iterator should be passed to this method since it's always called in a loop. It would shorten the calling code as it won't need to do for loops anymore.

Reply:
Just tried this out, and then reverted back. The problem is, sometimes we iterate over plain `spi_transaction_t` instances (`SpiBusDriver`). Sometimes, over `(spi_transaction_t, bool)` instances (`SpiDeviceDriver` / `SpiSoftCsDeviceDriver`). So we in fact need two variants of each of `spi_transmit` and `spi_transmit_async`, or we need to zip - in `SpiBusDriver` the stream of `spi_transaction_t` instances with another iterator generating `false`.

Doable, but IMO not worth the effort, and questionable in terms of LOCs or simplicity savings. Let's revisit once again when @Vollbrecht tries to upgrade the codebase to e-hal 1.0.alpha11 and to e-hal-async.